### PR TITLE
Added horizontal rule and task checkboxes

### PR DESCRIPTION
**Pull request by shabaev**
- Added [headlines.nvim](https://github.com/lukas-reineke/headlines.nvim) behave of horizontal rules;
- Improved [commit](https://github.com/JockeKv/markdown.nvim/commit/5fbcc4efdbbcc418c99a7e0ab654a90d5f7b2546) to be used in the plugin;
- Fixed tree-sitter query for the quotes.
```
> here is the first line
> here is the second line
>
> > here is the first line
> > here is the second line

> - [x] tasks#1
> > - [x] task#1.1
> > - [ ] task#1.2
```

### CODE REVIEW

**Review by MeanderingProgrammer — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -58,16 +60,33 @@ require('render-markdown').setup({
             (atx_h6_marker)
         ] @heading)
 
+        (thematic_break) @dash
```
[the remark this answers is not in the document]
Comment (MeanderingProgrammer) [reply README.md:63]: Merged the horizontal break change in: https://github.com/MeanderingProgrammer/markdown.nvim/commit/af819f39c63aeb09ff3801dbfd5188cea55e48e7

File: lua/render-markdown/handler/markdown.lua [hunk 2]
```diff
@@ -63,6 +63,27 @@ M.render = function(namespace, root)
...
                 virt_text_pos = 'overlay',
             })
+        elseif capture == 'dash' then
+            local width = vim.api.nvim_win_get_width(0)
+            local virt_text = { state.config.dash:rep(width), highlights.dash }
+            vim.api.nvim_buf_set_extmark(0, namespace, start_row, 0, {
+                virt_text = { virt_text },
+                virt_text_pos = 'overlay',
+            })
+        elseif capture == 'checkbox' then
```
[the remark this answers is not in the document]
Comment (shabaev) [reply lua/render-markdown/handler/markdown.lua]: no problem. done/checked - https://github.com/MeanderingProgrammer/markdown.nvim/pull/13/commits/9cca5b34fbf3fc366e81eed462f7368fd17bfbbe

File: README.md [hunk 3]
```diff
@@ -58,16 +60,33 @@ require('render-markdown').setup({
...
+          [
+            (task_list_marker_unchecked)
+            (task_list_marker_checked)
+          ]
+        ) @checkbox
 
-        (block_quote (block_quote_marker) @quote_marker)
-        (block_quote (paragraph (inline (block_continuation) @quote_marker)))
+        (block_quote_marker) @quote_marker
+        (block_continuation) @quote_marker
```
[the remark this answers is not in the document]
Comment (shabaev) [reply README.md:95]: Right you are. I typed `block_continuation` and meant kinda `block_quote_continuation`. Reverted the commit.
> Though I understand that this does not handle nested quotes particularly well.

Yup. Thanks.

**Comment by MeanderingProgrammer** [comment]
Thanks for putting this together!

I went ahead and merged the checkbox logic in this commit: https://github.com/MeanderingProgrammer/markdown.nvim/commit/90637a1120de47a3be57b00b7db4eee0d24834c8, largely copied from this PR.

Sorry to not just merge this PR as is, looks like you were in the process of making the final changes.

There were some modifications I made that I'll call out here:

* Everything in the `state.lua` Config class should be mandatory, the UserConfig makes everything optional but the actual Config should have a reasonable default for everything, including sub classes. This was probably just a copy pasta thing that left in the `?` in a few places.
* I wasn't sure why the task_list_marker_unchecked / task_list_marker_checked tree sitter queries included the list_marker_* prefix so I removed them and simplified those to the minimum. LMK if you had a reason for adding them. The reason to do this for the list_marker query was to avoid changing ordered lists, but since here we're updating the checkboxes, this isn't necessary.
* I removed the `(paragraph)` addition to the list_marker query, was also unsure about the reason for that, LMK if there was one.
* Slight change to how the condition for checkbox_unchecked & checkbox_checked is handled in `ui.lua` to reduce some of the duplication and handle different padding / character widths.

Some of these I might make more general and add to a contributing guide if there are more contributions, but I think it's getting to a fairly stable point.

Most are just my nit picky preferences.

I'll close out this PR since all the proposed changes are merged.

Feel free to shoot me any ideas or create more PRs more than happy to look over them.

Thanks again!

**Comment by shabaev** [comment]
gday,

> Sorry to not just merge this PR as is, looks like you were in the process of making the final changes.

nvm, it was a workable draft I were able and wanted to share with you to discuss and to pick some code. the goal achieved. thank you.

> Everything in the state.lua Config class should be mandatory,

yup. I overlooked that. thanks for watching my back :100: 

> included the list_marker_* prefix

the idea was to remove list bullets if it's a task list, but I agree with your point of view.

> to reduce some of the duplication

it looks perfect. thank you for the optimization (and to show me personally how to) :+1:

**Pull request closed without merging**